_posts/2018-03-08-roadmap.md

```diff
@@ -0,0 +1,221 @@
+---
+layout: post
+title: "Rust's 2018 roadmap"
+author: "Aaron Turon"
+---
+
+# Rust’s 2018 roadmap
+
+Each year the Rust community [comes together][roadmap-process] to set out a
+roadmap. This year, in addition to the [survey], we put out
+a [call for blog posts][blog-2018] in December, which resulted
+in [100 blog posts][read-rust] written over the span of a few weeks. The end
+result is the recently-merged [2018 roadmap RFC][rfc].
+
+[roadmap-process]: https://github.com/rust-lang/rfcs/pull/1728
+[survey]: https://blog.rust-lang.org/2017/09/05/Rust-2017-Survey-Results.html
+[blog-2018]: https://blog.rust-lang.org/2018/01/03/new-years-rust-a-call-for-community-blogposts.html
+[read-rust]: https://readrust.net/rust-2018/
+[rfc]: https://github.com/rust-lang/rfcs/pull/2314
+
+## Entering the Rust 2018 epoch
+
+**This year, we will deliver _Rust 2018_, marking the first new era in Rust programming since 1.0**.
+
+We will continue to publish releases every six weeks as usual. But we will
+designate a release in the latter third of the year (Rust 1.29 - 1.31) as the
+beginning of the Rust 2018 epoch. This “epoch release” will be the culmination
+of feature stabilization throughout the year, and will ship with polished
+documentation, tooling, and libraries that tie in to those features.
+
+The idea of “epochs” is to signify major steps in Rust’s evolution, where a
+collection of new features or idioms, taken as a whole, changes the experience
+of using Rust. They’re a chance, every few years, to take stock of the work
+we’ve delivered in six-week increments. To tell a bigger story about where Rust
+is going. And to ship the whole stack as a polished product.
+
+We expect that epochs will have a core theme or focus. Thinking of 1.0 as the
+start of the 2015 era, we have:
+
+-   Rust 2015: [stability](https://blog.rust-lang.org/2014/09/15/Rust-1.0.html)
+-   Rust 2018: productivity
+
+### Compatibility across epochs
+
+**TL;DR: Rust will continue its stability guarantee
+of [hassle-free updates to new versions][stability]**.
+
+[stability]: https://blog.rust-lang.org/2014/10/30/Stability.html
+
+Epochs will have a meaning for the compiler. You will be able to write:
+
+```toml
+epoch = "2018"
+```
+
+in your Cargo.toml to _opt in_ to the new epoch for your crate. Doing so may
```

Comment:
Did we decide on a flag for rustc? Otherwise I think we'll get pushback from people who want to not use cargo.

Reply:
cargo will indeed pass `-C epoch=2018` to rustc (currently `-Zepoch=2018`)